### PR TITLE
feat: class validator middleware for Hono

### PR DESCRIPTION
**Pull request by pr0m3th3usEx**
This middleware enables validation using [class-validator](https://github.com/typestack/class-validator) & [class-transformer](https://github.com/typestack/class-transformer) packages.

## Example

```ts
import { classValidator } from '@hono/class-validator'
import { IsInt, IsString } from 'class-validator'
 
class CreateUserDto {
  @IsString()
  name!: string;
 
  @IsInt()
  age!: number;
}
 
 
const route = app.post('/user', classValidator('json', CreateUserDto), (c) => {
  const user = c.req.valid('json')
  return c.json({ success: true, message: `${user.name} is ${user.age}` })
})
```

### CODE REVIEW

File: packages/class-validator/package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "type": "git",
+    "url": "https://github.com/honojs/middleware.git"
+  },
+  "homepage": "https://github.com/honojs/middleware",
+  "peerDependencies": {
+    "hono": ">=3.9.0"
+  },
+  "devDependencies": {
+    "hono": "^4.0.10",
+    "jest": "^29.7.0",
```
Comment (yusukebe) [inline packages/class-validator/package.json:37]: `jest` is not necessary.

File: packages/class-validator/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  "types": "dist/index.d.ts",
+  "exports": {
+    ".": {
+      "import": "./dist/index.js",
+      "types": "./dist/index.d.ts"
+    }
+  },
+  "scripts": {
+    "test": "vitest --run",
+    "build": "rimraf dist && tsc",
```
Comment (yusukebe) [inline packages/class-validator/package.json:18]: Can you use `tsup` to build? You can refer to this project: https://github.com/honojs/middleware/tree/main/packages/hello

File: .github/workflows/ci-class-validator.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+name: ci-cabin
```
Comment (yusukebe) [inline .github/workflows/ci-class-validator.yml:1]: This name should be renamed.

**Comment by yusukebe** [comment]
Hi @pr0m3th3usEx 

I've left the comments. Please check them!

**Comment by pr0m3th3usEx** [comment]
@yusukebe I made the changes

File: packages/auth-js/src/client.ts
Comment (yusukebe) [inline packages/auth-js/src/client.ts:1]: These packages such as `auth-js`, `event-emittter`, and `node-ws`, which are not related to `class-validator` should not be changed in this PR.

**Comment by pr0m3th3usEx** [comment]
@yusukebe Done

File: packages/class-validator/package.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "@hono/class-validator",
+  "packageManager": "yarn@4.0.2",
+  "description": "Validator middleware using Zod",
```
Comment (yusukebe) [inline packages/class-validator/package.json:4]: You have to change this description.

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@pr0m3th3usEx 

Looks good to me. Let's ship it! Thanks.

**Pull request merged**